src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -433,6 +532,342 @@ def can_create_clusterrolebindings(configuration):
+        from azure.storage.blob import generate_account_sas
+        if not is_valid_resource_id(storage_account):
+            storage_account_id = resource_id(
+                subscription=get_subscription_id(cli_ctx),
+                resource_group=rg_name,
+                namespace='Microsoft.Storage', type='storageAccounts',
+                name=storage_account
+            )
+        else:
+            storage_account_id = storage_account
+
+        if is_valid_resource_id(storage_account_id):
+            try:
+                parsed_storage_account = parse_resource_id(storage_account_id)
+            except Exception as ex:
+                logger.warning("Couldn't validate the storage account details. Error: {}".format(str(ex)))
+                return None, None, None
+        else:
+            logger.warning("Invalid storage account id - {}".format(storage_account_id))
+            return None, None, None
+
+        storage_account_name = parsed_storage_account['name']
+
+        readonly_sas_token = None
+        if sas_token is None:
+            storage_client = cf_storage(
+                cli_ctx, parsed_storage_account['subscription'])
+            storage_account_keys = storage_client.storage_accounts.list_keys(parsed_storage_account['resource_group'], storage_account_name)
+            sas_token = generate_account_sas(account_name=storage_account_name, account_key=storage_account_keys.keys[0].value, resource_types='sco', permission='rwdlacup', expiry=datetime.utcnow() + timedelta(days=1))
+            readonly_sas_token = generate_account_sas(account_name=storage_account_name, account_key=storage_account_keys.keys[0].value, resource_types='sco', permission='rl', expiry=datetime.utcnow() + timedelta(days=1))
+        return storage_account_name, sas_token, readonly_sas_token
+    except Exception as ex:
+        logger.warning("Error while validating the credentials for the storage account: {}".format(str(ex)))
+        return None, None, None
+
+
+def format_hyperlink(the_link):
+    return f'\033[1m{colorama.Style.BRIGHT}{colorama.Fore.BLUE}{the_link}{colorama.Style.RESET_ALL}'
+
+
+def format_bright(msg):
+    return f'\033[1m{colorama.Style.BRIGHT}{msg}{colorama.Style.RESET_ALL}'
+
+
+def display_diagnostics_report(kubectl_prior):   # pylint: disable=too-many-statements
+    if not which('kubectl'):
+        raise ValidationError('Can not find kubectl executable in PATH')
+    subprocess_cmd = kubectl_prior + ["get", "node", "--no-headers"]
+    nodes = subprocess.check_output(
+        subprocess_cmd,
+        universal_newlines=True)
+    logger.debug(nodes)
+    node_lines = nodes.splitlines()
+    ready_nodes = {}
+    for node_line in node_lines:
+        columns = node_line.split()
+        logger.debug(node_line)
+        if columns[1] != "Ready":
+            logger.warning("Node %s is not Ready. Current state is: %s.", columns[0], columns[1])
```

Comment:
@sophsoph321 Hey sophie, what exactly are these nodes that we're checking for? Are these the nodes which are already present as part of user's k8s cluster or is it something that periscope is adding?

Reply:
@sirireddy12 These are nodes which are already present as part of user's k8s cluster.